Issue #11 - Build dleyna inside the sandbox

hfiguiere opened the pull request.

hfiguiere commented:
no idea how to test this. This getting `(org.gnome.Photos:2): gnome-photos-WARNING **: 22:37:37.311: Unable to fetch the list of available renderers: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown`

hadess commented on org.gnome.Photos.json:
```diff
@@ -115,6 +114,74 @@
+            "name": "dleyna",
+            "buildsystem": "meson",
+            "config-opts": [
+                "-Dserver_name=com.intel.dleyna-renderer",
```
This is the wrong server name, you won't be able to own that. See the totem or rhythmbox manifests.

hfiguiere replied:
I made some incorrect assumptions. Fixed (I think)

hfiguiere commented:
Still getting the error:
```
C148: -> org.gnome.Photos.dleyna-renderer call com.intel.dLeynaRenderer.Manager.GetRenderers at /com/intel/dLeynaRenderer
B-1: <- org.freedesktop.DBus return from C146
*REWRITTEN*
B-1: <- org.freedesktop.DBus return from C147
*REWRITTEN*
B-1: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.ServiceUnknown from C148
B449: <- :1.1319 return from C145
C149: -> org.freedesktop.portal.Tracker call org.freedesktop.Tracker3.Endpoint.Query at /org/freedesktop/portal/Tracker/Session_32
B451: <- :1.1319 return from C149
C150: -> org.freedesktop.portal.Tracker call org.freedesktop.Tracker3.Endpoint.Query at /org/freedesktop/portal/Tracker/Session_32
B453: <- :1.1319 return from C150

(org.gnome.Photos:2): gnome-photos-WARNING **: 13:56:51.000: Unable to fetch the list of available renderers: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
```

Not sure how far it has to be patched.

hadess commented:
Sorry, I also forgot that in rhythmbox and totem's case, grilo also had a patch applied to be able to set the prefix for the dleyna DBus service, and the flatpak builds use that so the client and the server use that same prefix.

This is the grilo-plugins change: https://gitlab.gnome.org/GNOME/grilo-plugins/-/merge_requests/73

ebassi commented:
Photos is archived upstream, and EOL on Flathub.
